rshell/main.py

```diff
@@ -740,7 +740,10 @@ def get_stat(filename):
 def listdir(dirname):
     """Returns a list of filenames contained in the named directory."""
     import os
-    return os.listdir(dirname)
+    try:
+        return os.listdir(dirname)
+    except TypeError:
+        return os.listdir()
```

Comment:
It would be good to mention any known platforms which don't accept arguments.

Reply:
Atleast nrf pca10056.